packages/benchmark/benchmark/benchmark-liquidation.js

```diff
@@ -0,0 +1,233 @@
+// eslint-disable-next-line import/order
+import { bench } from '../src/benchmarkerator.js';
+
+import { Offers } from '@agoric/inter-protocol/src/clientSupport.js';
+import { scale6 } from '@agoric/boot/tools/liquidation.ts';
+
+const setupData = {
+  vaults: [
+    {
+      atom: 15,
+      ist: 100,
+      debt: 100.5,
+    },
+  ],
+  bids: [
+    {
+      give: '80IST',
+      discount: 0.1,
+    },
+  ],
+  price: {
+    starting: 12.34,
+    trigger: 9.99,
+  },
+  auction: {
+    start: {
+      collateral: 15,
+      debt: 100.5,
+    },
+    end: {
+      collateral: 9.659301,
+      debt: 0,
+    },
+  },
+};
+
+const outcome = {
+  bids: [
+    {
+      payouts: {
+        Bid: 0,
+        Collateral: 8.897786,
+      },
+    },
+  ],
+  reserve: {
+    allocations: {
+      ATOM: 6.102214,
+    },
+    shortfall: 20.5,
+  },
+  vaultsSpec: [
+    {
+      locked: 3.373,
+    },
+  ],
+  vaultsActual: [
+    {
+      locked: 3.525747,
+    },
+  ],
+};
+
+const liquidationOptions = {
+  collateralBrandKey: 'ATOM',
+  managerIndex: 0,
+};
+
+const setup = async context => {
+  const { managerIndex } = liquidationOptions;
+  const { walletFactoryDriver } = context.tools;
+
+  const metricsPath = `published.vaultFactory.managers.manager${managerIndex}.metrics`;
+  const buyer = await walletFactoryDriver.provideSmartWallet('agoric1buyer');
+  const minter = await walletFactoryDriver.provideSmartWallet('agoric1minter');
+
+  return { metricsPath, buyer, minter };
+};
+
+const setupRound = async (context, round) => {
+  const { collateralBrandKey, managerIndex } = liquidationOptions;
+  const { setupVaults, placeBids } = context.tools;
+
+  await setupVaults(
+    collateralBrandKey,
+    managerIndex,
+    setupData,
+    round * setupData.vaults.length,
+  );
+  await placeBids(collateralBrandKey, 'agoric1buyer', setupData);
+};
+
+const executeRound = async (context, round) => {
+  const { collateralBrandKey, managerIndex } = liquidationOptions;
+  const { advanceTimeBy, advanceTimeTo, priceFeedDrivers, readLatest } =
+    context.tools;
+
+  const { metricsPath } = context.config;
+
+  // ---------------
+  //  Change price to trigger liquidation
+  // ---------------
+
+  await priceFeedDrivers[collateralBrandKey].setPrice(9.99);
+
+  // check nothing liquidating yet
+  const liveSchedule = readLatest('published.auction.schedule');
```

Comment:
True. Since the vstorage is cached the difference will be small. Perhaps worth verifying.

Regardless, there can be a flag for whether to verify any values. Perhaps whether the final argument "t" is provided. That might be better than mocking it.

Reply:
Yes, though `readLatest` is reading from the simulated vstorage, which amounts to a `Map` lookup, so the overhead is modest.